httpx/_multipart.py

```diff
@@ -20,6 +20,16 @@
 )
 
 
+def get_multipart_boundary_from_content_type(
+    content_type: str,
+) -> bytes:
+    if ";" in content_type:
+        for section in content_type.split(";"):
+            if section.strip().startswith("boundary="):
+                return section.strip().split("boundary=")[-1].encode("latin-1")
+    raise ValueError("Missing boundary in multipart/form-data content-type header")
```

Comment:
Instead of raising `ValueError` we could just return `None` if the content type doesn't start with "multipart/form-data", or doesn't include a valid boundary.

(Users can currently submit requests with `Content-Type` headers that don't properly match up to the data they include, so just being lax here seems okay to me.)

Reply:
Note that since `content-type` gets set [via `setdefault`](https://github.com/encode/httpx/blob/aad60a4f123801e7ce0e02bc49138e7f0f9ca0a5/httpx/_models.py#L362). If it already exists, it won't be overwritten. So in the case where the user provided the `content-type` header but did not provide the multipart boundary we'd be sending out a request with no multipart boundary, which completely violates the spec and no server will be able to parse.

Reply:
> So in the case where the user provided the content-type header but did not provide the multipart boundary we'd be sending out a request with no multipart boundary, which completely violates the spec and no server will be able to parse.

True. We could either hard-error in that case, or just ignore it and allow users to do that.

There is also a whole class of cases here where a user can set a `Content-Type` that doesn't match the encoding type they're using with `data=`/`files=`/`json=`. (Eg. they can set `Content-Type: application/json` on either form or multipart requests.) But that's probably okay.